### PR TITLE
helper/resource: Show last state in timeout err message

### PR DESCRIPTION
**Pull request by radeksimko**
This PR is fixing a couple of existing tests that could easily report false positives and then adding a bit extra information into commonly seen error message to help the user to understand what the problem is.

Related: https://github.com/hashicorp/terraform/pull/8465

### CODE REVIEW

File: helper/resource/error.go [hunk 1]
```diff
@@ -40,15 +40,26 @@ func (e *UnexpectedStateError) Error() string {
...
 	if e.LastError != nil {
-		return fmt.Sprintf("timeout while waiting for state to become '%s': %s",
-			strings.Join(e.ExpectedState, ", "), e.LastError)
+		return fmt.Sprintf("timeout while waiting for %s%s: %s",
+			expectedState, lastState, e.LastError)
 	}
 
-	return fmt.Sprintf("timeout while waiting for state to become '%s'",
-		strings.Join(e.ExpectedState, ", "))
+	return fmt.Sprintf("timeout while waiting for %s%s",
```
Comment (stack72) [inline helper/resource/error.go]: Nit pick - should be a space between the %s values

File: helper/resource/error.go [hunk 2]
```diff
@@ -40,15 +40,26 @@ func (e *UnexpectedStateError) Error() string {
...
+
+	lastState := ""
+	if e.LastState != "" {
+		lastState = fmt.Sprintf(" (last state: '%s')", e.LastState)
+	}
+
 	if e.LastError != nil {
-		return fmt.Sprintf("timeout while waiting for state to become '%s': %s",
-			strings.Join(e.ExpectedState, ", "), e.LastError)
+		return fmt.Sprintf("timeout while waiting for %s%s: %s",
```
Comment (stack72) [inline helper/resource/error.go]: Nit pick - space required between %s values

**Comment by radeksimko** [comment]
@stack72 The `%s%s` is actually an intention to prevent the trailing space. See line 55 where there's an extra leading space. 😉

**Comment by stack72** [comment]
@radeksimko ah! in that case, ignore me :)

**Pull request merged**